src/Routing/Setter.purs

```diff
@@ -10,6 +10,14 @@ function setHash(hash) {
 }
 """ :: forall e. String -> Eff e Unit
 
+foreign import getHash """
+function getHash() {
+  return document.location.hash.replace(/^[^#]*#/g, "");
+}
+""" :: forall e. Eff e String
```

Comment:
This type signature is not accurate, because the hash can change. You need an effect type such as `DOM` or maybe a new one for Hash.